classes/submission/Collector.inc.php

```diff
@@ -55,8 +55,9 @@ abstract class Collector implements CollectorInterface
     public ?array $stageIds = null;
     public ?array $doiStatuses = null;
 
-    /** @var bool Whether null/empty values should count when considering Doi::STATUS_UNREGISTERED */
-    public bool $strictDoiStatusFilter = false;
+    public ?bool $hasDois = null;
+    /** @var array Which DOI types should be considered when checking if a submission has DOIs set */
+    public array $enabledDoiTypes = [];
```

Comment:
When there's a comment for a property, add an empty line above. (You can remove the empty line above $hasDois unless you want to add a comment.)